File: SmartCityIotDemo.Api/Entites/Device.cs
namespace SmartCityIotDemo.Api.Entities;
public class Device
{
    public int Id { get; set; }

    public string DeviceCode { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Location { get; set; } = string.Empty;

    public DateTime? LastSeenAt { get; set; }

    public bool IsOnline { get; set; }

    public List<TelemetryData> TelemetryData { get; set; } = new();
}

File: smartcity-dashboard/src/types/device.ts
export interface DeviceDashboard{
    deviceCode: string;
    name: string;
    location:string;
    isOnline: boolean;
    lastSeenAt: string | null;

    temperature: number | null;
    humidity: number | null;
    batteryLevel: number | null;
    signalStrength: number | null;

    telemetryCreatedAt: string | null;
}

export interface TelemetryMessage {
    deviceCode: string;
    temperature: number;
    humidity: number;
    batteryLevel: number;
    signalStrength: number;
    createdAt: string;
}

export interface AlarmMessage {
    deviceCode: string;
    alarmType: string;
    temperature: number;
    batteryLevel: number;
    createdAt: string;
}

